tests/components/UserActions/UserActions.spec.ts

```diff
@@ -417,4 +417,164 @@ describe('UserActions', () => {
+        let result: IQueryResult;
+        let resultElementRow: HTMLElement;
+        let resultElementCol: HTMLElement;
+        let viewedByCustomerOption: Boolean;
+
+        beforeEach(() => {
+            result = Fake.createFakeResult();
+
+            resultElementRow = document.createElement('div');
+            resultElementRow.classList.add('coveo-result-row');
+
+            resultElementCol = document.createElement('div');
+            resultElementCol.classList.add('coveo-result-cell');
+
+            resultElementRow.appendChild(resultElementCol);
+        });
+
+        describe('if the viewedByCustomer option is true', () => {
+            beforeEach(() => {
+                viewedByCustomerOption = true;
+            });
+
+            it('should add a ViewedByCustomer Component', () => {
+                const mock = Mock.advancedComponentSetup<UserActions>(
+                    UserActions,
+                    new Mock.AdvancedComponentSetupOptions(null, { userId: 'testUserId', viewedByCustomer: viewedByCustomerOption }, env => {
+                        fakeUserProfileModel(env.root, sandbox).getActions.returns(new Promise(() => {}));
+                        return env;
+                    })
+                );
+
+                const resultArgs = { result: result, item: resultElementRow };
+                Coveo.$$(mock.env.root).trigger(ResultListEvents.newResultDisplayed, resultArgs);
+
+                return delay(() => {
+                    expect(mock.cmp.options.viewedByCustomer).toBe(true);
+                    expect(resultElementRow.getElementsByClassName('CoveoViewedByCustomer').length).toBe(1);
+                });
+            });
+
+            it('It should not add a viewedByCustomer if one is already there', () => {
+                const mock = Mock.advancedComponentSetup<UserActions>(
+                    UserActions,
+                    new Mock.AdvancedComponentSetupOptions(null, { userId: 'testUserId', viewedByCustomer: viewedByCustomerOption }, env => {
+                        fakeUserProfileModel(env.root, sandbox).getActions.returns(new Promise(() => {}));
+                        return env;
+                    })
+                );
+
+                resultElementRow.querySelector('.coveo-result-cell').classList.add('CoveoViewedByCustomer');
+
+                const resultArgs = { result: result, item: resultElementRow };
+                Coveo.$$(mock.env.root).trigger(ResultListEvents.newResultDisplayed, resultArgs);
+
+                return delay(() => {
+                    expect(resultElementRow.getElementsByClassName('CoveoViewedByCustomer').length).toBe(1);
+                });
+            });
+        });
+
+        describe('if the viewedByCustomer option is false', () => {
+            beforeEach(() => {
+                viewedByCustomerOption = false;
+            });
+            it('should not add a ViewedByCustomer Component', () => {
```

Comment:
Breakline between 539 and 540, check for good spacing like this across this file :)